### ISSUE TITLE
QAFTestStep not logged to test-results (custom TestNG framework).

### ISSUE THREAD

**Issue by mialeska** [qmetry/qaf#445]
### QAF Version
Latest release - 3.2.0
> Note: only the latest version is supported
We have existing maven solution with large amount of selenium tests written in java with testNG. Now we would like to import execution results into QMetry for Jira. 
We decided to integrate QAF package into our existing solution, to get results in qas format, with failure screenshots and steps included. 

As a starter, I followed the example in your repository: https://github.com/qmetry/qtm4j-automation-sample-projects/blob/master/QAF/pom.xml a
I managed to set our driver instance to qaf, so currently we have got test results with screenshots. But now I still have an issue with it:

I started marking our steps (we call them pageActions) with @QAFTestStep annotations, to see them as checkpoints in the generated test result jsons. I have also added resources/application.properties with the parameter step.provider.pkg=pageActions (pageActions is a full name of our package with steps. I have added related plugins and dependencies for aspectj and qaf into our pom.xml . I can see that aspects has been generated (AjcClosure classes generated in  target directory after the build). But unfortunately steps are not getting added to test result jsons. Could you please suggest where could be the problem or how to achieve what we want? 

### Steps To Reproduce
    1.  Add QAF dependencies to maven project
    2. Extend BaseTest from WebDriverTestCase and initialize browser at TestBaseProvider in beforeMethod:    
```
        TestBaseProvider.instance().get()
                .setDriver("chrome", new QAFExtendedWebDriver(DriverManager.getDriver()));
```
    3. Add @QAFTestStep annotation to desired test steps
    4. Run the maven command mvn clean test

### Expected behavior
Method marked with @QAFTestStep annotation gets added as checkpoint to test-results/ ... .json file

### Actual behavior
Only errors and assertions are present in  test-results/ ... .json file

### Is the issue reproducible on runner?

- [] QAS
- [x] Maven
- [ ] Gradle
- [] Ant
- [ ] Eclipse

### Test case sample
I prepared a temp repository with sample maven project with our dependencies, where the issue can be reproduced:
https://github.com/mialeska/tmp-ui-tests/tree/test-results

Please find generated test results in the test-results branch.

**Comment by cjayswal** [comment]
Refer [pom file](https://github.com/qmetry/qaf-blank-project-maven/blob/master/pom.xml) from sample maven project.

**Comment by mialeska** [comment]
@cjayswal yes I used it for the reference, but add some customizations for project specific:
- we are working with IntelliJ IDEA, not eclipse
- we already have some libraries that have aspectJ and some other dependencies 
Could you please point out what in particular you do see as wrong with my [pom](https://github.com/mialeska/tmp-ui-tests/blob/test-results/pom.xml) file?

**Comment by cjayswal** [comment]
Maven aspectj-maven-plugin is not specific to eclipse. If you don't want to use `aspectj-maven-plugin` refer [configuration](https://github.com/qmetry/qaf-blank-project-maven/blob/master/pom.xml#L81) used in plugin.

**Comment by mialeska** [comment]
@cjayswal I am using the aspectj-maven-plugin. But my configuration is a little different, as with yours config **Lombok** dependency is not working.
Lombok requires aspectJ to be configured in the maven-surefire-plugin's argline:
```
      <argLine>
            -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
      </argLine>
```
With your configuration added, I see the following error:


[WARNING] You aren't using a compiler supported by lombok, so lombok will not work and has been disabled.
Your processor is: org.aspectj.org.eclipse.jdt.internal.compiler.apt.dispatch.BatchProcessingEnvImpl
Lombok supports: OpenJDK javac, ECJ
	`<unknown source file>:<no line information>`
	
[WARNING] advice defined in com.qmetry.qaf.automation.step.JavaStepReporter has not been applied [Xlint:adviceDidNotMatch]
	C:\Users\a2.meleshko\.m2\repository\com\qmetry\qaf\3.2.0\qaf-3.2.0.jar!com\qmetry\qaf\automation\step\JavaStepReporter.class:0
	


Below is my configuration for plugins section:

```
       <plugins>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>aspectj-maven-plugin</artifactId>
                <version>1.14.0</version>
                <executions>
                    <execution>
                        <id>default-compile</id>
                        <phase>process-classes</phase>
                        <goals>
                            <goal>compile</goal>
                        </goals>
                        <configuration>
                            <weaveDirectories>
                                <weaveDirectory>${project.build.directory}/classes</weaveDirectory>
                            </weaveDirectories>
                        </configuration>
                    </execution>
                    <execution>
                        <id>default-testCompile</id>
                        <phase>process-test-classes</phase>
                        <goals>
                            <goal>test-compile</goal>
                        </goals>
                        <configuration>
                            <weaveDirectories>
                                <weaveDirectory>${project.build.directory}/test-classes</weaveDirectory>
                            </weaveDirectories>
                        </configuration>
                    </execution>
                </executions>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                    <showWeaveInfo>true</showWeaveInfo>
                    <verbose>true</verbose>
                    <complianceLevel>1.8</complianceLevel>
                    <aspectLibraries>
                        <aspectLibrary>
                            <groupId>com.qmetry</groupId>
                            <artifactId>qaf</artifactId>
                        </aspectLibrary>
                    </aspectLibraries>
                    <sources/>
                    <excludes>
                        <exclude>**/*.java</exclude>
                    </excludes>
                    <forceAjcCompile>true</forceAjcCompile>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>org.aspectj</groupId>
                        <artifactId>aspectjtools</artifactId>
                        <version>${aspectj.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
        </plugins>
```
